### PR TITLE
Resolved: Disable DocumentClientResourceLeakTest #188

### PR DESCRIPTION
**Pull request by David-Noble-at-work**
#188

### CODE REVIEW

File: benchmark/pom.xml [hunk 1]
```diff
@@ -107,89 +107,71 @@
...
+      <groupId>io.dropwizard.metrics</groupId>
+      <artifactId>metrics-jvm</artifactId>
+      <version>${metrics.version}</version>
     </dependency>
     <dependency>
-      <groupId>org.slf4j</groupId>
-      <artifactId>slf4j-api</artifactId>
-      <version>${slf4j.version}</version>
+      <groupId>io.netty</groupId>
+      <artifactId>netty-tcnative</artifactId>
```
Comment (christopheranderson) [inline benchmark/pom.xml]: Apache 2.0 License. Approved. ✅ [Link](https://github.com/netty/netty-tcnative/blob/master/LICENSE.txt)

File: examples/src/test/java/com/microsoft/azure/cosmosdb/rx/examples/UniqueIndexAsyncAPITest.java [hunk 2]
```diff
@@ -47,15 +48,17 @@
...
     private final static int TIMEOUT = 60000;
 
     private AsyncDocumentClient client;
     private Database createdDatabase;
 
     @Test(groups = "samples", timeOut = TIMEOUT)
     public void uniqueIndex() {
-        DocumentCollection collectionDefinition = new DocumentCollection();
+
+        DocumentCollection collectionDefinition = new DocumentCollection();//
```
Comment (moderakh) [inline examples/src/test/java/com/microsoft/azure/cosmosdb/rx/examples/UniqueIndexAsyncAPITest.java]: please remove the tailing "//"

File: sdk/src/test/java/com/microsoft/azure/cosmosdb/rx/DocumentClientResourceLeakTest.java [hunk 3]
```diff
@@ -39,49 +37,52 @@
...
     private DocumentCollection createdCollection;
 
     @Factory(dataProvider = "simpleClientBuildersWithDirect")
     public DocumentClientResourceLeakTest(Builder clientBuilder) {
-        this.clientBuilder = clientBuilder;
+        super(clientBuilder);
     }
 
-    @Test(groups = {"emulator"}, timeOut = TIMEOUT)
+    @Test(enabled = false, groups = {"emulator"}, timeOut = TIMEOUT)
```
Comment (moderakh) [inline sdk/src/test/java/com/microsoft/azure/cosmosdb/rx/DocumentClientResourceLeakTest.java]: if this only fails for TCP, can we only disable this for TCP? and let it run for https

File: benchmark/src/test/java/com/microsoft/azure/cosmosdb/benchmark/ReadMyWritesConsistencyTest.java [hunk 4]
```diff
@@ -56,55 +57,64 @@
...
     private final int delayForInitiationCollectionScaleUpInSeconds = 60;
     private final Duration defaultMaxRunningTimeInSeconds = Duration.ofMinutes(45);
 
-    private final String maxRunningTime =
-                    System.getProperty("MAX_RUNNING_TIME", StringUtils.defaultString(Strings.emptyToNull(
-                            System.getenv().get("MAX_RUNNING_TIME")), defaultMaxRunningTimeInSeconds.toString()));
-
     private final AtomicBoolean collectionScaleUpFailed = new AtomicBoolean(false);
+
+    private final String directModeProtocol = System.getProperty("cosmos.directModeProtocol", Protocol.Tcp.name());
```
[the remark this answers is not in the document]
Comment (moderakh) [reply benchmark/src/test/java/com/microsoft/azure/cosmosdb/benchmark/ReadMyWritesConsistencyTest.java]: this one discussed offline.

File: benchmark/src/test/java/com/microsoft/azure/cosmosdb/benchmark/ReadMyWritesConsistencyTest.java [hunk 5]
```diff
@@ -56,55 +57,64 @@
...
+            collection.getId(),
+            desiredConsistency,
+            concurrency,
+            numberOfOperationsAsString,
+            maxRunningTime
+        ) + (useNameLink ? " -useNameLink" : "");
 
         Configuration cfg = new Configuration();
-        new JCommander(cfg, StringUtils.split(cmd));
+        JCommander commander = new JCommander(cfg, StringUtils.split(cmd));
```
[the remark this answers is not in the document]
Comment (David-Noble-at-work) [reply benchmark/src/test/java/com/microsoft/azure/cosmosdb/benchmark/ReadMyWritesConsistencyTest.java]: I've revised this on my current current working branches:

- danoble/rntbd/issue-119
- danoble/rntbd/v3

**Comment by christopheranderson** [comment]
Merged this manually due to merge conflicts - https://github.com/Azure/azure-cosmosdb-java/commit/95fa58efa19cf1d5ac5ed67e30bce30e91e34917

**Pull request closed without merging**